### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by shemalucien**
### **What does this PR do?**

- Implements  Sign up and login Authentication 

- Implements input validation 

### Description of Task to be completed?

- User can be able to sign in and login with validated inputs

### How should this be manually tested?

- Clone the repo
- checkout to branch authentication
- Run npm install to install modules
- Run npm run build and nodemon dist/index.js to start the development server

### What are the relevant Pivotal Tracker/trello stories?

- link to my pivotal tracker/https://trello.com/b/TssCCVAb/my-portfollio-backend

### Screenshots (if appropriate)
-N/A
### Questions:

- N/A

### CODE REVIEW

File: dist/database/index.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+// mongoose.connect('mongodb+srv://shemalucien:Shema123@articles.2pt7v.mongodb.net/todo-app-db', {
+//     useNewUrlParser: true,
+//     useUnifiedTopology: true
+// })
+//     .then(() => {
+//         console.log("App connected to Mongodb successfully")
+//     })
+//     .catch((e) => {
+//         console.log("Mongodb connection error " + e.message);
+//     })
```
Comment (PatrickNiyogitare28) [inline dist/database/index.js:16]: Remove the comments. Not good to push them on production

File: dist/controllers/query.controller.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
+"use strict";
+
+Object.defineProperty(exports, "__esModule", {
+  value: true
+});
+exports.updateQuery = exports.saveQuery = exports.getById = exports.getAllQueries = exports.deleteQueryById = void 0;
+
+var _query = _interopRequireDefault(require("../database/model/query.model"));
+
```
Comment (PatrickNiyogitare28) [inline dist/controllers/query.controller.js:9]: Replace `var` with `let` for our code quality and practice of using latest js syntax

File: dist/database/model/query.model.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+"use strict";
+
+Object.defineProperty(exports, "__esModule", {
+  value: true
+});
+exports.default = void 0;
+
+var _mongoose = _interopRequireDefault(require("mongoose"));
+
```
Comment (PatrickNiyogitare28) [inline dist/database/model/query.model.js:9]: use `let` instead of `var`

File: dist/index.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+"use strict";
+
+require("./database");
+
+var _express = _interopRequireDefault(require("express"));
+
+var _query = _interopRequireDefault(require("./routes/query.route"));
```
Comment (PatrickNiyogitare28) [inline dist/index.js:7]: use `let` instead of `var`

File: dist/routes/query.route.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+"use strict";
+
+Object.defineProperty(exports, "__esModule", {
+  value: true
+});
+exports.default = void 0;
+
+var _express = _interopRequireDefault(require("express"));
+
+var _query = require("../controllers/query.controller");
```
Comment (PatrickNiyogitare28) [inline dist/routes/query.route.js:10]: replace `var` with `let`

File: dist/routes/query.route.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+const router = _express.default.Router();
+
+router.post('/', _query.saveQuery);
+router.get('/', _query.getAllQueries);
+router.get('/:id', _query.getById);
+router.put('/:id', _query.updateQuery);
+router.delete('/:id', _query.deleteQueryById);
+var _default = router;
+exports.default = _default;
```
Comment (PatrickNiyogitare28) [inline dist/routes/query.route.js:22]: replace `var` with `let`

File: src/controllers/auth.controller.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+import { decodeToken, signToken } from '../helpers/jwt';
+
+export const signup = async (req, res) => {
+    let user = req.body;
+    user.password = await hash(user.password);
+
+    // Validate user input
+    // if (!(user.email && user.password && user.firstName && user.lastName)) {
+    //     res.status(400).send("All input is required");
+    // }
```
Comment (PatrickNiyogitare28) [inline src/controllers/auth.controller.js:12]: Remove comments

**Review by PatrickNiyogitare28 — CHANGES_REQUESTED** [review]
Work on the bellow code quality-based feedback after pushing the changes please let me know. so that we can approve this.

File: src/controllers/auth.controller.js (hunk 7 above)
Comment (shemalucien) [reply src/controllers/auth.controller.js:12]: thanks patrick i will let you know after correction

**Review by PrinceNiyonshuti — APPROVED** [review]
nice work , and keep working on making your codes clean by removing the commented one

**Review by PatrickNiyogitare28 — APPROVED** [review]
Interesting, we're good to go now. You can merge

**Pull request merged**